ibis/backends/sql/rewrites.py

```diff
@@ -236,9 +236,14 @@ def sqlize(
     simplified = sqlized.replace(merge_select_select)
 
     # extract common table expressions while wrapping them in a CTE node
-    ctes = extract_ctes(simplified)
-    subs = {cte: CTE(cte) for cte in ctes}
-    result = simplified.replace(subs)
+    ctes = set(extract_ctes(simplified))
```

Comment:
```suggestion
    ctes = frozenset(extract_ctes(simplified))
```

Might as well make it immutable.

Reply:
Going to update.